src/ServicePulse.Host/app/js/views/custom_checks/customChecks.controller.js

```diff
@@ -34,12 +35,19 @@
             $scope.loadingData = true;
             $scope.model = { data: [], total: 0 };
             $scope.disableLoadingData = false;
-            load(page);
+            $scope.reloadCount++;
+
+            load(page, $scope.reloadCount);
         }
 
-        function load(page) {
+        function load(page, reloadCount) {
             serviceControlService.getFailingCustomChecks(page).then(function (response) {
 
+                //If the load was called before last result reload it should be discarded
```

Comment:
Should this decrement the page number as well?

Reply:
I don't think so. Whenever `reloadData` is called the page number is [set back to 1](https://github.com/Particular/ServicePulse/pull/931/files#diff-97d95757170a4ea9d39b4fae0f079725R34). So the next load caused by the scroll will start from 2. All reloads caused by the scroll that were done when before latest refresh will get discarded.